gateway/modules/crud/bolt/update.go

```diff
@@ -0,0 +1,95 @@
+package bolt
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+
+	"github.com/boltdb/bolt"
+	"github.com/sirupsen/logrus"
+
+	"github.com/spaceuptech/space-cloud/gateway/model"
+	"github.com/spaceuptech/space-cloud/gateway/utils"
+)
+
+func (b *Bolt) Update(ctx context.Context, project, col string, req *model.UpdateRequest) (int64, error) {
+
+	switch req.Operation {
+	case utils.One, utils.Upsert:
+		// Update single document
+		if err := b.client.Update(func(tx *bolt.Tx) error {
+			// Assume bucket exists and has keys
+			bucket := tx.Bucket([]byte(project))
+			c := bucket.Cursor()
+
+			// get all keys matching the prefix
+			prefix := []byte(col + "/")
+			for k, v := c.Seek(prefix); k != nil && bytes.HasPrefix(k, prefix); k, v = c.Next() {
+				result := map[string]interface{}{}
+				if err := json.Unmarshal(v, result); err != nil {
+					logrus.Errorf("error un marshalling while reading from bboltdb - %v", err)
+					return err
+				}
+				// if valid then update
+				if utils.Validate(req.Find, result) {
+					value, err := json.Marshal(req.Update)
```

Comment:
req,Update contains a map like the following:
```json
{
  "$set": {"name": "John"}
}
```
You need to selectively update fields in the current object, and overwrite using the updated object